poc/tests/test_vidpf.py

```diff
@@ -90,3 +91,154 @@ def test(self):
+                    prefixes,
+                    nonce,
+                )
+                proofs.append(proof)
+            self.assertFalse(vidpf.verify(proofs[0], proofs[1]))
+
+    def test_malformed_correction_word_ctrl(self):
+        vidpf = Vidpf(Field128, 5, 1)
+        nonce = gen_rand(vidpf.NONCE_SIZE)
+        rand = gen_rand(vidpf.RAND_SIZE)
+        (public_share, keys) = vidpf.gen(0, [Field128(1)], nonce, rand)
+
+        # Tweak some control bit of some correction word.
+        malformed_level = randrange(vidpf.BITS)
+        (seed_cw, ctrl_cw, w_cw, proof_cw) = public_share[malformed_level]
+        malformed = ctrl_cw.copy()
+        malformed[randrange(2)] += Field2(1)
+        public_share[malformed_level] = (seed_cw, malformed, w_cw, proof_cw)
+
+        # The tweak doesn't impact the computation until we reach the level
+        # with the malformed correction word.
+        for level in range(malformed_level, vidpf.BITS):
+            prefixes = tuple(range(2**level))
+            proofs = []
+            for agg_id in range(2):
+                (_beta_share, _out_share, proof) = vidpf.eval(
+                    agg_id,
+                    public_share,
+                    keys[agg_id],
+                    level,
+                    prefixes,
+                    nonce,
+                )
+                proofs.append(proof)
+            self.assertFalse(vidpf.verify(proofs[0], proofs[1]))
+
+    def test_malformed_correction_word_payload(self):
+        vidpf = Vidpf(Field128, 5, 1)
+        nonce = gen_rand(vidpf.NONCE_SIZE)
+        rand = gen_rand(vidpf.RAND_SIZE)
+        (public_share, keys) = vidpf.gen(0, [Field128(1)], nonce, rand)
+
+        # Tweak the payload of some correction word.
+        malformed_level = randrange(vidpf.BITS)
+        (seed_cw, ctrl_cw, w_cw, proof_cw) = public_share[malformed_level]
+        malformed = w_cw.copy()
+        malformed[randrange(vidpf.VALUE_LEN)] += Field128(1)
+        public_share[malformed_level] = (seed_cw, ctrl_cw, malformed, proof_cw)
+
+        # The tweak doesn't impact the computation until we reach the level
+        # with the malformed correction word.
+        for level in range(malformed_level, vidpf.BITS):
+            prefixes = tuple(range(2**level))
+            proofs = []
+            for agg_id in range(2):
+                (_beta_share, _out_share, proof) = vidpf.eval(
+                    agg_id,
+                    public_share,
+                    keys[agg_id],
+                    level,
+                    prefixes,
+                    nonce,
+                )
+                proofs.append(proof)
+            self.assertFalse(vidpf.verify(proofs[0], proofs[1]))
+
+    # TODO Figure out we expect the proof to be malleable or if there's a bug
+    # in our code. This test demonstrates that we can tweak the proof of a
+    # correction word without being detected.
+    @unittest.skip("this test is known to fail")
```

Comment:
@jimouris any insights here?

Reply:
Alrighty, I've widdled this down a bit, but there's still somewhat of a gap in my understanding. See https://github.com/jimouris/draft-mouris-cfrg-mastic/pull/66/files#diff-403a3f1b2b91292a0217bda576e71e402ba0ffce878b0e44ad56fcffc0cbf83fR236-R261.